tests/test_util.py

```diff
@@ -66,27 +67,109 @@ def test_clone_git_repo_raises_explicit_exception_on_clone_fail(
 
 
 class OutputTests(unittest.TestCase):
+    @mock.patch("tartufo.scanner.ScannerBase")
+    @mock.patch("tartufo.types.GlobalOptions")
     @mock.patch("tartufo.util.click")
-    def test_echo_issues_echos_all_when_not_json(self, mock_click):
-        util.echo_issues([1, 2, 3, 4], False, "", "")
+    def test_echo_result_echos_all_when_not_json(
+        self, mock_click, mock_options, mock_scanner
+    ):
+        mock_options.json = False
+        mock_options.verbose = 0
+        mock_scanner.exclude_signatures = []
+        mock_scanner.issues = [1, 2, 3, 4]
+        util.echo_result(mock_options, mock_scanner, "", "")
         mock_click.echo.assert_has_calls(
             (mock.call(1), mock.call(2), mock.call(3), mock.call(4)), any_order=False
         )
 
+    @mock.patch("tartufo.scanner.ScannerBase")
+    @mock.patch("tartufo.types.GlobalOptions")
+    @mock.patch("tartufo.util.click")
+    @mock.patch("tartufo.util.datetime")
+    def test_echo_result_echos_message_when_clean(
+        self, mock_time, mock_click, mock_options, mock_scanner
+    ):
+        mock_time.now.return_value.isoformat.return_value = "now:now:now"
+        mock_options.json = False
+        mock_options.quiet = False
+        mock_options.verbose = 0
+        mock_scanner.exclude_signatures = []
+        mock_scanner.issues = []
+        util.echo_result(mock_options, mock_scanner, "", "")
+        mock_click.echo.assert_called_with(
+            "Time: now:now:now\nAll clear. No secrets detected."
+        )
+
+    @mock.patch("tartufo.scanner.ScannerBase")
+    @mock.patch("tartufo.types.GlobalOptions")
+    @mock.patch("tartufo.util.click")
+    @mock.patch("tartufo.util.datetime")
+    def test_echo_result_echos_exclusions_verbose(
+        self, mock_time, mock_click, mock_options, mock_scanner
+    ):
+        mock_time.now.return_value.isoformat.return_value = "now:now:now"
+        mock_options.json = False
+        mock_options.quiet = False
+        mock_options.verbose = 1
+        mock_scanner.issues = []
+        mock_scanner.excluded_paths = [
+            re.compile("package-lock.json"),
+            re.compile("poetry.lock"),
+        ]
+        mock_options.exclude_signatures = [
+            "fffffffffffff",
+            "ooooooooooooo",
+        ]
+        util.echo_result(mock_options, mock_scanner, "", "")
+        mock_click.echo.assert_has_calls(
+            (
+                mock.call("Time: now:now:now\nAll clear. No secrets detected."),
+                mock.call("\nExcluded paths:"),
+                mock.call("package-lock.json\npoetry.lock"),
+                mock.call("\nExcluded signatures:"),
+                mock.call("fffffffffffff\nooooooooooooo"),
```

Comment:
Scanning over this quickly, I didn't realize this was "\n" and just read it as "noooooooooooo" 🤣 

Well played.